src/phoenix/server/api/types/ProjectSession.py

```diff
@@ -1,20 +1,128 @@
-from typing import Optional
+from datetime import datetime
+from typing import TYPE_CHECKING, Annotated, ClassVar, Optional, Type
 
 import strawberry
-from sqlalchemy import desc, select
-from strawberry import UNSET, Info
+from openinference.semconv.trace import SpanAttributes
+from sqlalchemy import func, select
+from sqlalchemy.sql.functions import coalesce
+from strawberry import UNSET, Info, lazy
 from strawberry.relay import Connection, Node, NodeID
 
 from phoenix.db import models
 from phoenix.server.api.context import Context
+from phoenix.server.api.types.MimeType import MimeType
 from phoenix.server.api.types.pagination import ConnectionArgs, CursorString, connection_from_list
-from phoenix.server.api.types.Trace import Trace, to_gql_trace
+from phoenix.server.api.types.SpanIOValue import SpanIOValue
+from phoenix.server.api.types.TokenUsage import TokenUsage
+
+if TYPE_CHECKING:
+    from phoenix.server.api.types.Trace import Trace
 
 
 @strawberry.type
 class ProjectSession(Node):
+    _table: ClassVar[Type[models.ProjectSession]] = models.ProjectSession
     id_attr: NodeID[int]
     session_id: str
+    session_user: Optional[str]
+    start_time: datetime
+    end_time: datetime
+
+    @strawberry.field(description="Duration of the session in seconds")
+    async def duration(
+        self,
+        info: Info[Context, None],
+    ) -> int:
+        return int((self.end_time - self.start_time).total_seconds())
+
+    @strawberry.field
+    async def num_traces(
+        self,
+        info: Info[Context, None],
+    ) -> int:
+        stmt = select(func.count(models.Trace.id)).filter_by(project_session_rowid=self.id_attr)
+        async with info.context.db() as session:
+            return await session.scalar(stmt) or 0
+
+    @strawberry.field
+    async def first_input(
+        self,
+        info: Info[Context, None],
+    ) -> Optional[SpanIOValue]:
+        stmt = (
+            select(
+                models.Span.attributes[INPUT_VALUE].label("value"),
+                models.Span.attributes[INPUT_MIME_TYPE].label("mime_type"),
```

Comment:
Non-blocking: We might want to think about materializing the input and output as their own columns so we can avoid querying attributes over a connection. Even with a dataloader, this led to performance issues with traces when the attributes are large.